### PR TITLE
Multiple Roots - complete implementation

### LINKED ISSUE #117: Parsing all types in a file with *
Currently I use the following command to generate a schema:
`ts-json-schema-generator --path test.ts --type MyInterface`

However, instead of just generating a schema containing `MyInterface`, is it possible to collect all types within `test.ts` and write them in to the same schema?

### PR DESCRIPTION
**Pull request by andreashuber69**
- [x] Adapt types to the reality that Config.type can be undefined for the implementation of #117
- [x] Add tests
- [x] Update README

Fixes #117

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,16 +20,15 @@ This project is made possible by a [community of contributors](https://github.co
 
 ```bash
 npm install --save ts-json-schema-generator
-./node_modules/.bin/ts-json-schema-generator \
-    --path 'my/project/**.*.ts' \
-    --type 'My.Type.Full.Name' \
-    --expose 'export' \
-    --jsDoc 'extended'
+./node_modules/.bin/ts-json-schema-generator --path 'my/project/**.*.ts'
```
Comment (domoritz) [inline README.md]: The typical usage will be to provide a type so let's leave that here in the docs.

Comment (andreashuber69) [reply README.md]: > The typical usage will be to provide a type so let's leave that here in the docs.

Agreed.

**Comment by domoritz** [comment]
Thank you!

**Comment by andreashuber69** [comment]
> Thank you!

You're welcome. Thank you for maintaining this project!

**Pull request merged**